index.js

```diff
@@ -51,5 +51,155 @@ const state = {
       price: 0.35
     }
   ],
-  cart: []
-};
\ No newline at end of file
+  cart: [],
+  total: [],
+};
+
+// Function to add item to cart
+function addToCart(index) {
+  
+  //new item
+  const newItem = state.items[index];
+
+  //check if item already exists
+  const existingItemIndex = state.cart.findIndex(item => item.id === newItem.id);
+
+  if (existingItemIndex !== -1) 
+  { //increase quantity if item is already in cart
+    state.cart[existingItemIndex].quantity++;
+  } else
+  { //otherwise add to cart with quantity = 1
+    newItem.quantity = 1;
+    state.cart.push(newItem);
+  }
+
+  //increase total price
+  state.total.push(newItem.price);
+  const sum = state.total.reduce((total, currentValue) => 
+  total + currentValue, 0);
+  renderCart() //update cart
+}
+
+//function to increase quantity
+function increaseQuantity(item) {
+  item.quantity++;
+  state.total.push(item.price);
+  renderCart(); //update the cart display
+}
+
+//function to decrease quantity
+function decreaseQuantity(item) {
+  if (item.quantity === 1)
+  {
+    state.cart.pop(item);
```

Comment:
When you're trying to remove an item from the cart or decrease its quantity, using pop() is not the correct approach because pop() removes the last element of an array regardless of which item you're targeting. Instead, you should filter the cart for the specific item. Also, using pop() on state.total doesn't specify which price to remove.

Reply:
Resolved